ctl/src/command.rs

```diff
@@ -1293,34 +1278,30 @@ pub fn query_application(mut channel: Channel<CommandRequest,CommandResponse>, j
 
   match channel.read_message() {
     None          => {
-      eprintln!("the proxy didn't answer");
-      exit(1);
+      bail!("the proxy didn't answer");
     },
     Some(message) => {
       if id != message.id {
-        eprintln!("received message with invalid id: {:?}", message);
-        exit(1);
+        bail!("received message with invalid id: {:?}", message);
       }
       match message.status {
         CommandStatus::Processing => {
           // do nothing here
           // for other messages, we would loop over read_message
           // until an error or ok message was sent
+          Ok(())
         },
         CommandStatus::Error => {
           if json {
-            print_json_response(&message.message);
-          } else {
-            eprintln!("could not query proxy state: {}", message.message);
+            print_json_response(&message.message)?;
           }
-          exit(1);
+          bail!("could not query proxy state: {}", message.message);
         },
         CommandStatus::Ok => {
           if let Some(needle) = application_id.or(domain) {
             if let Some(CommandResponseData::Query(data)) = message.data {
               if json {
-                print_json_response(&data);
-                return;
+                return Ok(print_json_response(&data)?);
```

Comment:
This is bad practice. Better would be because it is clear to see that the return is in fact just an emtpy tuple:

```
                print_json_response(&data)?;
                return Ok(());
```